### PR TITLE
Client side of state bags for weapons

### PR DESCRIPTION
**Pull request by Fusionfivem**

### CODE REVIEW

File: [core]/es_extended/client/main.lua [hunk 1]
```diff
@@ -358,8 +358,11 @@ function StartServerSyncLoops()
 						else
 							if ammoCount ~= currentWeapon.Ammo then
 								currentWeapon.Ammo = ammoCount
-								TriggerServerEvent('esx:updateWeaponAmmo', weapon.name, ammoCount)
-							end
+                            Player(GetPlayerServerId(PlayerId())).state:set('ESX::WeaponUpdate', {
```
Comment (Gellipapa) [inline [core]/es_extended/client/main.lua:361]: I think add variable.

File: [core]/es_extended/client/main.lua [hunk 2]
```diff
@@ -358,8 +358,11 @@ function StartServerSyncLoops()
...
+								name = weapon.name,
+								ammo = ammoCount, 
+							    Random = GetGameTimer() + math.random(99999999, 999999999)
```
Comment (Gellipapa) [inline [core]/es_extended/client/main.lua:364]: What is this?

Comment (Fusionfivem) [reply [core]/es_extended/client/main.lua:364]: state bags instead of triggering the server event way more optimized

**Comment by TheFantomas** [comment]
Make only one pr...

File: [core]/es_extended/client/main.lua (hunk 2 above)
Comment (Gellipapa) [reply [core]/es_extended/client/main.lua:364]: Sorry i mean this row: 
`Random = GetGameTimer() + math.random(99999999, 999999999)
 `

Comment (Rav3n95) [reply [core]/es_extended/client/main.lua:364]: LocalPlayer.state:set ??????????

Comment (Fusionfivem) [reply [core]/es_extended/client/main.lua:364]: To what gellipapa said it doesnt pass half the time without to what Rav3n Said im not the brightest

Comment (Gellipapa) [reply [core]/es_extended/client/main.lua:364]: Sorry i don't understand what do you mean. Raven tell you you should use LocalPlayer in client side. Read FiveM docs how to works statebags.

Comment (Fusionfivem) [reply [core]/es_extended/client/main.lua:364]: LocalPlayer.state:set('ESX::WeaponUpdate', {
								name = weapon.name,
								ammo = ammoCount, 
							    Random = GetGameTimer() + math.random(99999999, 999999999)

Comment (Fusionfivem) [reply [core]/es_extended/client/main.lua:364]: fixed

Comment (CsokiHUN) [reply [core]/es_extended/client/main.lua:364]: > fixed

Please use what Raven said or we will not accept it.

> LocalPlayer.state:set ??????????

**Pull request closed without merging**